planner/core/integration_test.go

```diff
@@ -4954,6 +4954,32 @@ func (s *testIntegrationSuite) TestIssue30094(c *C) {
 	))
 }
 
+func (s *testIntegrationSuite) TestIssue30200(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t1;")
+	tk.MustExec("create table t1(c1 varchar(100), c2 varchar(100), key(c1), key(c2), c3 varchar(100));")
+	tk.MustExec("insert into t1 values('ab', '10', '10');")
+	// lpad has not been pushed to TiKV or TiFlash.
+	tk.MustQuery("explain select /*+ use_index_merge(t1) */ * from t1 where c1 = 'ab' or c2 = '10' and char_length(lpad(c1, 10, 'a')) = 10;").Check(testkit.Rows(
```

Comment:
Better to use `explain format=brief ...`.

Reply:
done